### ISSUE TITLE
'!' flag doesn't add space in "no-prefix" case

### ISSUE THREAD

**Issue by polyanskiy** [Rockhopper-Technologies/prefixed#4]
Very useful package!
I am using it with a '!' flag to display an output of a joulemeter with a space between the number and the unit for better readability. This works fine except the 1--1000 "no-prefix" range: in this case no space is added.
```
f'{Float(0.123):!.3H}J'
'123 mJ'
f'{Float(123):!.3H}J'
'123J'
f'{Float(123000):!.3H}J'
'123 kJ'
```

**Comment by avylove** [comment]
@polyanskiy, thanks for reporting this.

You're right, I can see  where that is happening and why you would want the space. I think this was an oversight on my part.

The problem I have is how to handle a case where a space is not desired. For example, if you're not adding anything after the prefix. Ideally, both cases could be handled intuitively.  I'd like to think on it before I implement something. My currently thought is a double flag `!!` could specify to trim the space when no prefix is listed, but I'm not sure if that would be consistent with the regular Python [spec](https://docs.python.org/3/library/string.html#formatspec).

What are your thoughts on it?

**Comment by polyanskiy** [comment]
hmm... I don't see how to handle this without a different flag (or a double flag as you suggest).
... Actually, are there any realistic situation when the space is needed other than for separating the unit from the number? I mean, when a prefix is used to simply shorten the number (e.g., "$1M"), the space is never used.

**Comment by avylove** [comment]
You never really know how someone is going to use a library. It's best to be flexible and non-opinionated.
Regardless, I think the behavior for `!` should be consistent. I pushed a commit that fixes the behavior. Please confirm it works for you. I'll do a release soon.

**Comment by avylove** [comment]
Fixed and added `!!` flag in [0.6.0](https://github.com/Rockhopper-Technologies/prefixed/releases/tag/0.6.0). Thanks for reporting!
